src/doublycircularlist/DoublyCircularLinkedList.java

```diff
@@ -0,0 +1,186 @@
+package doublycircularlist;
+
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.LinkedList;
+
+public class DoublyCircularLinkedList <E>{
+
+    private Node first;
```

Comment:
As a best practice you should always use the generic when declaring an object. Like:
private Node<E> first;
And the same advice can be applied to all the file. E.g the add method.